src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.ReadObjectResponse.liquid

```diff
@@ -2,37 +2,83 @@
 
 protected virtual async System.Threading.Tasks.Task<ObjectResponseResult<T>> ReadObjectResponseAsync<T>(System.Net.Http.HttpResponseMessage response, System.Collections.Generic.IReadOnlyDictionary<string, System.Collections.Generic.IEnumerable<string>> headers)
 {
+    if (response.Content == null)
+    {
+        return new ObjectResponseResult<T>(default(T), string.Empty);
+    }
+
     if (ReadResponseAsString)
     {
         var responseText = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-        try
+
+        if (response.Content.Headers.ContentType != null && response.Content.Headers.ContentType.MediaType.Contains("xml"))
```

Comment:
What should we check here?

Reply:
I think it looks ok, the common is probably application/xml, but this covers text/xml also.
The idea here is only to support the basic 1-1 with C# class and XML?

Reply:
If the `response.Content == null`, shouldn't the `ResponseText` be `null` too instead of `String.Empty`?

Reply:
`response.Content.Headers.ContentType.MediaType.Contains("xml"))` - this is probably incorrect. The OOXML formats are XML files stored in a PKZIP container but have MIME types like `application/vnd.openxmlformats-officedocument.wordprocessingml.document`.

I think opting for a whitelist that users can override is better:

    public virtual bool IsXml( string mimeContentType ) {
        switch( mimeContentType ) {
        case "application/xml":
        case "application/xhtml+xml":
        case "text/xml":
            return true;
        default:
            //return false;
            return
                mimeContentType.EndsWith( "+xml", StringComparison.Ordinal )  ||
                mimeContentType.EndsWith( "/xml", StringComparison.Ordinal ); // risky?
        }
    }